### PR TITLE
47 implement token enrollment via validatecheck

### PR DESCRIPTION
**Pull request by lukasmatusiewicz**

### CODE REVIEW

File: src/test/java/org/privacyidea/TestPollTransaction.java [hunk 1]
```diff
@@ -100,6 +101,19 @@ public void testPushSynchronous() throws InterruptedException
         assertEquals("push", pushChallenge.getType());
         assertTrue(pushChallenge.getAttributes().isEmpty());
 
+        String imagePush = "";
+        for (Challenge c : challenges)
+        {
+            if (Objects.equals(c.getType(), "push"))
```
Comment (nilsbehlen) [inline src/test/java/org/privacyidea/TestPollTransaction.java:107]: "push".equals(c.getType())

File: src/test/java/org/privacyidea/TestTriggerChallenge.java [hunk 2]
```diff
@@ -103,6 +106,20 @@ public void testTriggerChallengeSuccess()
         // result
         assertTrue(responseTriggerChallenge.status);
         assertFalse(responseTriggerChallenge.value);
+
+        List<Challenge> challenges = responseTriggerChallenge.multichallenge;
+        String imageTOTP = "";
+        for (Challenge c : challenges)
+        {
+            if (Objects.equals(c.getType(), "totp"))
```
Comment (nilsbehlen) [inline src/test/java/org/privacyidea/TestTriggerChallenge.java:114]: see above

**Review by nilsbehlen — CHANGES_REQUESTED** [review]
also please remove the imports after changing the equals usage

File: src/test/java/org/privacyidea/TestTriggerChallenge.java (hunk 2 above)
Comment (lukasmatusiewicz) [reply src/test/java/org/privacyidea/TestTriggerChallenge.java]: [dd2fb58](https://github.com/privacyidea/java-client/pull/52/commits/dd2fb581273914ba37982dbf761599e80770d195)

File: src/test/java/org/privacyidea/TestPollTransaction.java (hunk 1 above)
Comment (lukasmatusiewicz) [reply src/test/java/org/privacyidea/TestPollTransaction.java]: [dd2fb58](https://github.com/privacyidea/java-client/pull/52/commits/dd2fb581273914ba37982dbf761599e80770d195)

**Review by nilsbehlen — APPROVED** [review]

**Pull request merged**